File: mobile/android/app/src/main/java/com/findly/android/ui/designsystem/token/SpacingTokens.kt
package com.findly.android.ui.designsystem.token

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * The spacing scale (specs/003-android-client.md §4.1/§4.2).
 *
 * Kotlin identifier note: the semantic token name is `2xl` (per the shared cross-platform
 * vocabulary in the A1 task brief) but `2xl` is not a legal Kotlin identifier (cannot start with
 * a digit). The property below is named [xxl] and MUST be treated as the same token as `2xl` by
 * any future design-generation tool targeting this contract.
 */
@Immutable
data class FindlySpacingTokens(
    val xs: Dp,
    val sm: Dp,
    val md: Dp,
    val lg: Dp,
    val xl: Dp,
    val xxl: Dp,
)

// Design 2a — "Ember / Dusk" (design/findly-design-system/2a-ember-dusk/HANDOFF.md).
val FindlySpacing = FindlySpacingTokens(
    xs = 4.dp,
    sm = 8.dp,
    md = 12.dp,
    lg = 20.dp,
    xl = 28.dp,
    xxl = 40.dp,
)

val LocalFindlySpacing = staticCompositionLocalOf { FindlySpacing }

File: mobile/ios/FindlyKit/Sources/FindlyKit/DesignSystem/Tokens/SpacingTokens.swift
import CoreGraphics

/// specs/004-ios-client.md §2.1 — spacing scale in points.
public struct SpacingTokens: Equatable {
    public var xs: CGFloat
    public var sm: CGFloat
    public var md: CGFloat
    public var lg: CGFloat
    public var xl: CGFloat
    public var xxl: CGFloat

    public init(xs: CGFloat, sm: CGFloat, md: CGFloat, lg: CGFloat, xl: CGFloat, xxl: CGFloat) {
        self.xs = xs
        self.sm = sm
        self.md = md
        self.lg = lg
        self.xl = xl
        self.xxl = xxl
    }

    // design 2a "Ember/Dusk" (design/findly-design-system/2a-ember-dusk/HANDOFF.md).
    public static let standard = SpacingTokens(xs: 4, sm: 8, md: 12, lg: 20, xl: 28, xxl: 40)
}
